timdex_dataset_api/embeddings.py

```diff
@@ -106,10 +126,145 @@ def __init__(self, timdex_dataset: TIMDEXDataset):
         self.schema = TIMDEX_DATASET_EMBEDDINGS_SCHEMA
         self.partition_columns = ["year", "month", "day"]
 
+        # DuckDB context
+        self.conn = self.setup_duckdb_context()
+        self._sa_metadata_data_schema = sa_reflect_duckdb_conn(self.conn, schema="data")
+
+        # resolve data type for 'embedding_vector' column
+        if "data.embeddings" in self._sa_metadata_data_schema.tables:
+            sa_metadata_data_embeddings_table = self._sa_metadata_data_schema.tables[
+                "data.embeddings"
+            ]
+            sa_metadata_data_embeddings_table.c.embedding_vector.type = ARRAY(FLOAT)
+
     @property
     def data_embeddings_root(self) -> str:
         return f"{self.timdex_dataset.location.removesuffix('/')}/data/embeddings"
 
+    def get_sa_table(self, table: str) -> Table:
+        """Get SQLAlchemy Table from reflected SQLAlchemy metadata."""
+        schema_table = f"data.{table}"
+        if schema_table not in self._sa_metadata_data_schema.tables:
+            raise ValueError(f"Could not find table '{table}' in DuckDB schema 'data'.")
+        return self._sa_metadata_data_schema.tables[schema_table]
+
+    def setup_duckdb_context(self) -> DuckDBPyConnection:
+        """Create a DuckDB connection for embeddings query and retrieval.
+
+        This method extends TIMDEXDatasetMetadata's pre-existing DuckDB connection
+        (via the attached TIMDEXDataset), creating views in the 'data' schema.
+        """
+        start_time = time.perf_counter()
+
+        conn = self.timdex_dataset.conn
+
+        try:
+            self._create_embeddings_view(conn)
+            self._create_current_embeddings_view(conn)
+            self._create_current_run_embeddings_view(conn)
+        except DuckDBIOException:
+            logger.warning("No embeddings found")
+        except Exception as exception:  # noqa: BLE001
+            logger.warning(f"An error occurred while creating views: {exception}")
+
+        logger.debug(
+            f"DuckDB data context created, {round(time.perf_counter()-start_time,2)}s"
+        )
```

Comment:
Similar story here to a comment below (this was added later), where I think this logging message could be improved.  

Perhaps we should say something like,

```
DEBUG:DuckDB context created for TIMDEXEmbeddings
```  

Then we could also update `TIMDEXDataset` and `TIMDEXDatasetMetadata` as well.  It'd be clear when each class method finished, _what_ DuckDB context had completed.

Reply:
Thanks for the suggestion!